src/ol/geom/circle.js

```diff
@@ -223,7 +222,7 @@ ol.geom.Circle.prototype.setFlatCoordinates = function(layout, flatCoordinates)
  * @api
  */
 ol.geom.Circle.prototype.setRadius = function(radius) {
-  goog.asserts.assert(this.flatCoordinates,
+  ol.DEBUG && console.assert(this.flatCoordinates,
```

Comment:
Some of these asserts are a bit gratuitous.  If somehow we deleted `this.flatCoordinates`, calling `setRadius()` would result in this:
```
circle.js:225 Assertion failed: truthy this.flatCoordinates expectedol.geom.Circle.setRadius @ circle.js:225(anonymous function) @ VM1746:1
circle.js:227 Uncaught TypeError: Cannot read property '0' of undefined(…)
```

Here the statement immediately after the `console.assert()` throws under the same condition that the assert arg is truthy.  So, without the assert, we would get an unhandled error with a message and stack trace identifying the problem.  In this case, I don't think there is a lot of value in the assert.

In other cases, there is value.  For example, if someone calls `setCenter()` above with a center that has a different length than the original center, things break later in mysterious ways (there may be no error thrown at all).

You've done a ton of work here already, but maybe later we could go back and remove this type of assert (where an error would be thrown in the next statement (or very nearby) in cases where the assert arg is truthy).